docs/html/user_guide.rst

```diff
@@ -1304,6 +1304,229 @@ issue tracker`_ if you believe that your problem has exposed a bug in pip.
+
+In the case where a package has a lot of versions, arriving at a good
+candidate can take a lot of time. (The amount of time depends on the
+package size, the number of versions pip must try, and other concerns.)
+
+How does backtracking work?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When doing a pip install, it needs to start by making assumptions about the
+packages it needs to install. During the install process it needs check this
+assumptions as it goes along.
+
+When it finds that an assumption is incorrect, it has to try another approach
+(backtrack), which means discarding some of the work that has already been done,
+and going back to choose another path.
+
+For example; The user requests ``pip install tea``. ```tea`` has dependencies of
+``cup``, ``hot-water``, ``spoon`` amongst others.
+
+pip starts by installing a version of ``cup``. If it finds out it isn’t
+compatible (with the other package versions) it needs to “go back”
+(backtrack) and download an older version.
+
+It then tries to install that version. If it is successful, it will continue
+onto the next package. If not it will continue to backtrack until it finds a
+compatible version.
+
+This backtrack behaviour can end in 2 ways - either 1) it will
+successfully find a set packages it can install (good news!), or 2) it will
+eventually display `resolution impossible <https://pip.pypa.io/en/latest/user_guide/#id35>`__ error
+message (not so good).
+
+If pip starts backtracking during dependency resolution, it does not
+know how long it will backtrack, and how much computation would be
+needed. For the user this means it can take a long time to complete.
+
+Why does backtracking occur?
+----------------------------
+
+With the release of the new resolver (:ref:`Resolver changes 2020`), pip is now
+more strict in the package versions it installs when a users runs a
+``pip install`` command.
+
+Pip needs to backtrack because initially, it doesn't have all the information it
+needs to work out the correct set of packages. This is because package indexes
+don't provide full package dependency information before you have downloaded
+the package.
+
+This new resolver behaviour means that pip works harder to find out which
+version of a package is a good candidate to install. It reduces the risk that
+installing a new package will accidentally break an existing installed package,
+and so reducing the risk of your environment gets messed up.
+
+What does this behaviour look like?
+-----------------------------------
+
+Right now backtracking behaviour looks like this:
+
+::
+
+   $ pip install tea==1.9.8
+   Collecting tea==1.9.8
+   Downloading tea-1.9.8-py2.py3-none-any.whl (346 kB)
+   |████████████████████████████████| 346 kB 10.4 MB/s
+   Collecting spoon==2.27.0
+   Downloading spoon-2.27.0-py2.py3-none-any.whl (312 kB)
+   |████████████████████████████████| 312 kB 19.2 MB/s
+   Collecting hot-water>=0.1.9
+   Downloading hot-water-0.1.13-py3-none-any.whl (9.3 kB)
```

Comment:
I think we should fix the indentation here.

Reply:
Can you tell me specifically what needs to be fixed? Do you mean outdent/dedent it?

Reply:
This example does not indent the contents (like the Downloading lines) while pip's output does:

```
$ pip install pytest
Collecting pytest
  Downloading pytest-6.1.1-py3-none-any.whl (272 kB)
     |████████████████████████████████| 272 kB 3.6 MB/s 
Collecting pluggy<1.0,>=0.12
  Downloading pluggy-0.13.1-py2.py3-none-any.whl (18 kB)
Requirement already satisfied: packaging in /Users/pradyunsg/.virtualenvs/pip/lib/python3.8/site-packages (from pytest) (20.4)
Collecting iniconfig
  Downloading iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)
Collecting attrs>=17.4.0
  Downloading attrs-20.2.0-py2.py3-none-any.whl (48 kB)
     |████████████████████████████████| 48 kB 10.7 MB/s 
Requirement already satisfied: toml in /Users/pradyunsg/.virtualenvs/pip/lib/python3.8/site-packages (from pytest) (0.10.1)
Collecting py>=1.8.2
  Downloading py-1.9.0-py2.py3-none-any.whl (99 kB)
     |████████████████████████████████| 99 kB 4.0 MB/s 
Requirement already satisfied: six in /Users/pradyunsg/.virtualenvs/pip/lib/python3.8/site-packages (from packaging->pytest) (1.15.0)
Requirement already satisfied: pyparsing>=2.0.2 in /Users/pradyunsg/.virtualenvs/pip/lib/python3.8/site-packages (from packaging->pytest) (2.4.7)
Installing collected packages: pluggy, iniconfig, attrs, py, pytest
Successfully installed attrs-20.2.0 iniconfig-1.1.1 pluggy-0.13.1 py-1.9.0 pytest-6.1.1
```